app/models/proposal.rb

```diff
@@ -30,6 +30,8 @@ class Proposal < ActiveRecord::Base
 
   acts_as_taggable
 
+  # we do not have "dependent:" declaration on "steps" because there can be circular references
```

Comment:
would it be possible to write a test that illustrates how the cascade works rather than having this comment here? 

I have some idea of what it means, but I am not exactly sure what the "circular references" you refer to are, so a test would help me to understand more clearly.